api/src/main/java/module-info.java

```diff
@@ -636,15 +636,15 @@
  * <tr style="vertical-align: top; background-color:#eee"><td>{@code countBy...}</td>
  * <td>{@code long},
  * <br>{@code int}</td>
- * <td>Jakarta Persistence providers limit the maximum to {@code Integer.MAX_VALUE}</td></tr>
```

Comment:
I know I wrote these statements at one point, and I thought I remember doing so in response to limitations that I found in the Jakarta Persistence spec but I cannot find it now, and in any case, it doesn't make any sense to me why the language was on methods that are returning counts rather than find methods that return lists of results, which seems the more likely place to encounter limits.

Reply:
Ah, right, well it's true that a Java `List` or array can only have `MAX_INTEGER` elements, but that's not specific to Jakarta Persistence. Anyway, that many entities in memory will definitely blow up your computer.